pandas/core/categorical.py

```diff
@@ -1995,14 +1995,18 @@ def is_dtype_equal(self, other):
 
         Parameters
         ----------
-        other : Categorical
+        other : Categorical, Series
 
         Returns
         -------
         are_equal : boolean
         """
-
         try:
+            from pandas.core.series import Series
+
+            if isinstance(other, Series):
```

Comment:
Ah, that makes sense I guess.

```
In [5]: pd.Categorical(pd.Categorical([0, 1], ordered=True))
Out[5]:
[0, 1]
Categories (2, int64): [0, 1]
```

so not ordered. The default for the constructor is `ordered=False`.

I would say just explicitly check for the known cases of `Categorical`, `Series` w/ category dtype, and `CategoricalIndex` and just extract the `categories` and `ordered` off of them and build a new `other = Categorical(categories, ordered=ordered)`.

Once I finish https://github.com/pandas-dev/pandas/pull/16015 this will be much simpler (just use `.dtype`), but that won't be for a few weeks probably.